restricted_site_access.php

```diff
@@ -1529,26 +1529,99 @@ public static function ip_in_range( $ip, $range ) {
 	 * @return string
 	 */
 	public static function get_client_ip_address() {
-		$ip      = '';
-		$headers = array(
+		// REMOTE_ADDR IP address.
+		$remote_addr_header_ip = isset( $_SERVER['REMOTE_ADDR'] ) ? sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ) ) : false;
+
+		// Return if REMOTE_ADDR is not set.
+		if ( empty( $remote_addr_header_ip ) ) {
+			return '';
+		}
+
+		/*
+		 * Filter hook to set array of trusted proxies.
+		 *
+		 * Some reverse proxies (like AWS Elastic Load Balancing) don't have
+		 * a static IP address or even a range that you can target with the CIDR notation.
+		 * In this case, you'll need to - very carefully - trust all proxies by setting
+		 * $trusted_proxies to an empty array - (default behaviour).
+		 *
+		 * In case your reverse proxy uses static IP addresses, then you can add those
+		 * addresses to the $trusted_proxies array.
+		 *
+		 * @param string[] $trusted_proxies The IP addresses of the proxy we want to trust.
+		 */
+		$trusted_proxies = apply_filters( 'rsa_trusted_proxies', array() );
+
+		if ( ! empty( $trusted_proxies ) ) {
```

Comment:
The `empty()` prevents developers disallowing any proxies. I'd expect the following to prevent proxy access:

```php
add_filter( 'rsa_trusted_proxies', '__return_empty_array' );
```

It might be worht checking with `has_filter()` but I could be convinced otherwise.

Reply:
Just want to make sure I'm following correctly. You're suggesting we replace the `empty` check with a `has_filter` check, allowing someone to use this filter to return an empty array, which would then pass our check but because there's no proxies to iterate over at that point, we would just return an empty string which should in essence block access to the site?

Guess I'm just wondering what the use case here would be for this, seems like this could be achieved already using the settings in the admin. I'm not opposed to making this change just having a hard time coming up with a scenario where this would be helpful.

Reply:
It's possible I am misunderstanding the purpose of the code. Let me do some testing during the day and get back to you. If needs be I will provide a test to clarify my thinking.

Reply:
I had myself totally confused when I read this code yesterday. I see what it's doing and it all makes sense to me.